### ISSUE TITLE
[CRITICAL] [VIP-Travel] Create workspace address page and section in overview

### ISSUE THREAD

**Issue by stitesExpensify** [Expensify/App#37827]
## VIP-Travel Implementation Issue
Tracking Issue: https://github.com/Expensify/Expensify/issues/333471
Project: https://github.com/orgs/Expensify/projects/112/
Relevant Design Doc Section: https://docs.google.com/document/d/122DLLTuCsU4NC3c7rNiy4UF8bZxVlLEWSy3yncqCTcQ/edit#bookmark=id.9gtuxny77xql

This issue is for creating the new workspace address page:
<img width="799" alt="2024-03-06_08-42-12" src="https://github.com/Expensify/App/assets/42391420/151e2f87-246f-4102-b4c6-520ef1f241bf">

- A new MenuItemWithTopDescription will be added to [WorkspaceOverviewPage](https://github.com/Expensify/App/blob/cf8279f28ca69e84bf0001c7d8206e1037ef014a/src/pages/workspace/WorkspaceOverviewPage.js#L120) to display the Company address as shown above. On click, it’ll open WorkspaceOverviewAddressPage.
- A new page, WorkspaceOverviewAddressPage will be created, with [a new route](https://docs.google.com/document/d/122DLLTuCsU4NC3c7rNiy4UF8bZxVlLEWSy3yncqCTcQ/edit#bookmark=id.gn6jbl2udlnx), ROUTES.WORKSPACE_PROFILE_ADDRESS. It will mostly mimic [AddressPage.js](https://github.com/Expensify/App/blob/main/src/pages/settings/Profile/PersonalDetails/AddressPage.js) with a couple of changes:
- Instead of receiving the [user’s personal details](https://github.com/Expensify/App/blob/a0614e9f68f1c066958f481b22559d9bc37e7d99/src/pages/settings/Profile/PersonalDetails/AddressPage.js#L21), it’ll receive the workspace data, including the address.
- Instead of calling [PersonalDetails.UpdateAddress](https://github.com/Expensify/App/blob/a0614e9f68f1c066958f481b22559d9bc37e7d99/src/pages/settings/Profile/PersonalDetails/AddressPage.js#L59), it will call a new function, Policy.updateAddress.
Policy.updateAddress will behave much like [Policy.updateWorkspaceCustomUnitAndRate](https://github.com/Expensify/App/blob/a0614e9f68f1c066958f481b22559d9bc37e7d99/src/libs/actions/Policy.ts#L935), but it will call a new command, UpdateWorkspaceAddress.
- The PHP layer will be updated to receive the new command in [api.php](https://github.com/Expensify/Web-Expensify/blob/main/api.php), and it will call a new function, PolicyAPI::updateWorkspaceAddress, which, in turn, will call a new Auth command.. [WIP]
- The current [Policy.updateGeneralSettings](https://github.com/Expensify/App/blob/a0614e9f68f1c066958f481b22559d9bc37e7d99/src/libs/actions/Policy.ts#L804)… [WIP]

Routes table:
https://docs.google.com/document/d/122DLLTuCsU4NC3c7rNiy4UF8bZxVlLEWSy3yncqCTcQ/edit#bookmark=id.gn6jbl2udlnx

**Comment by smelaa** [comment]
Hello, I'm Aleksandra from Software Mansion, the expert agency, and I'd like to work on this issue.

**Comment by twisterdotcom** [comment]
Adding a C+ here for Travel reviews: @shubham1206agra

**Comment by twisterdotcom** [comment]
Only assigned on Friday. We can give them a day here Melvin.

**Comment by smelaa** [comment]
Hi,

> Instead of receiving the [user’s personal details](https://github.com/Expensify/App/blob/a0614e9f68f1c066958f481b22559d9bc37e7d99/src/pages/settings/Profile/PersonalDetails/AddressPage.js#L21), it’ll receive the workspace data, including the address.

Could you please let me know under which prop can I find that address? Is it already implemented?

**Comment by twisterdotcom** [comment]
Draft PR above.

**Comment by smelaa** [comment]
Would you me to make the changes visible only for Spotnana Travel beta users?

**Comment by stitesExpensify** [comment]
Yes please!

**Comment by twisterdotcom** [comment]
How is this one coming along @smelaa?

**Comment by twisterdotcom** [comment]
Still waiting on webhooks.

**Comment by twisterdotcom** [comment]
Waiting on the backend.

**Comment by twisterdotcom** [comment]
@stitesExpensify @deetergp we don't need to wait for webhooks for this one do we? We're just saving this somewhere we always know where it is.

**Comment by stitesExpensify** [comment]
Correct we just need this API endpoint https://github.com/Expensify/Expensify/issues/385614#issuecomment-2057322402

**Comment by twisterdotcom** [comment]
This is unblocked now right? Let's get this one moving @smelaa!

**Comment by twisterdotcom** [comment]
Can we make this ready for review yet @smelaa?

**Comment by smelaa** [comment]
@stitesExpensify @twisterdotcom Can I have more details on this [API endpoint](https://github.com/Expensify/App/issues/37827#issuecomment-2057404685)? I think the linked issue is internal and I can't see the link.

**Comment by Gonals** [comment]
Discussed it via DM, but posting here for visibility.

For the endPoint, it requires 2 parameters (pesides the AuthToken):

- policyID

- data, which is an array with the address containing:

```
addressStreet (the street name and number, like "1234 Any Street". Decondary address line, if any, should be separated by a newline (\n) character)

city

country

state

zipCode
```

**Comment by twisterdotcom** [comment]
Update: https://expensify.slack.com/archives/C05S5EV2JTX/p1714085472583259?thread_ts=1713825690.019579&cid=C05S5EV2JTX

**Comment by lakchote** [comment]
cc @smelaa looks like a lot of blockers are related to it

**Comment by twisterdotcom** [comment]
This is already live!

**Comment by twisterdotcom** [comment]
Ah! Didn't realise this was linked to a PR, thought it was one of those handled in another issue. Found it: https://github.com/Expensify/App/pull/38381

**Comment by twisterdotcom** [comment]
Payment Summary:

- @smelaa does not require payment (Contractor)
- @shubham1206agra paid $250 [here](https://www.upwork.com/nx/wm/offer/102320635) (C+)
